openedx_tagging/core/tagging/rest_api/v1/serializers.py

```diff
@@ -175,3 +176,33 @@ class TaxonomyTagDeleteBodySerializer(serializers.Serializer):  # pylint: disabl
         child=serializers.CharField(), required=True
     )
     with_subtags = serializers.BooleanField(required=False)
+
+
+class TaxonomyImportBodySerializer(serializers.Serializer):  # pylint: disable=abstract-method
+    """
+    Serializer of the body for the Taxonomy Import action
+    """
+    taxonomy_name = serializers.CharField(required=True)
+    taxonomy_description = serializers.CharField(default="")
+    file = serializers.FileField(required=True)
+
+    def get_parser_format(self, obj):
+        """
+        Returns the ParserFormat based on the file extension
+        """
+        filename = obj["file"].name
+        ext = filename.split('.')[-1]
+        if ext.lower() == 'csv':
+            return ParserFormat.CSV
+        elif ext.lower() == 'json':
+            return ParserFormat.JSON
+        else:
+            raise serializers.ValidationError(f'File type not supported: ${ext.lower()}')
+
+    def to_internal_value(self, data):
+        """
+        Adds the parser_format to the validated data
+        """
+        validated_data = super().to_internal_value(data)
+        validated_data['parser_format'] = self.get_parser_format(data)
+        return validated_data
```

Comment:
You can avoid having to override `to_internal_value` this by adding a `SerializerMethodField` for the `parser_format` field.

ref https://www.django-rest-framework.org/api-guide/fields/#serializermethodfield

```suggestion
```

Reply:
I tried this, but it didn't work. I think it works when serializing, not deserializing. The value is filled in the `to_representation` [method](https://github.com/encode/django-rest-framework/blob/f56b85b7dd7e4f786e0769bba6b7609d4507da83/rest_framework/fields.py#L1869-L1871), not in the `to_internal_value`.

Reply:
Ach, bummer.

But can we avoid the duplicated code in `validate_file` and `get_parser_format`?

Reply:
I'm cheating here or is it okay to do it on the `validate`? d400a5418557b3622d9178be54c4de75372a90f6